switch_model/transmission/transport/build.py

```diff
@@ -193,18 +193,25 @@ def define_components(mod):
         'trans_length_km', 'trans_efficiency', 'existing_trans_cap')
     mod.trans_new_build_allowed = Param(
         mod.TRANSMISSION_LINES, within=Boolean, default=True)
+    mod.trans_capital_cost_per_mw_km = Param(
+        within=NonNegativeReals,
+        default=1000)
     mod.TRANS_BLD_YRS = Set(
         dimen=2,
         initialize=mod.TRANSMISSION_LINES * mod.PERIODS,
-        filter=lambda m, tx, p: m.trans_new_build_allowed[tx])
+        filter=lambda m, tx, p: m.trans_new_build_allowed[tx] and m.trans_capital_cost_per_mw_km != float("inf"))
```

Comment:
Help me understand this. We are just filtering the set if the Tx is allowed to build and does not have infinitive cost?

Reply:
Exactly. TRANS_BLD_YRS should only be years where the transmission can be built so we want to filter out when that's not allowed. I added a catch for infinite cost since if the cost is infinite we don't want to allow building.